src/Hakyll/Links.hs

```diff
@@ -0,0 +1,56 @@
+module Hakyll.Links
+       ( applyShortcuts
+       , applyAllShortcuts
+       , shortcutLinksCompiler
+       , allShortcutLinksCompiler
+       ) where
+
+import Data.Maybe (fromMaybe)
+import Data.Text (Text)
+import Hakyll (Compiler, Item, defaultHakyllReaderOptions, defaultHakyllWriterOptions,
+               pandocCompilerWithTransform)
+import ShortcutLinks (Result (..), Shortcut, allShortcuts, useShortcutFrom)
+import Text.Pandoc.Generic (bottomUp)
+
+import Hakyll.Links.Parser (parseShortcut)
+
+import qualified Data.Text as T
+import qualified Text.Pandoc.Definition as Pandoc
+
+
+{- | Modifies links to add an extra anchor which links to the header.
+-}
+applyShortcuts :: [([Text], Shortcut)] -> Pandoc.Pandoc -> Pandoc.Pandoc
+applyShortcuts shortcuts = bottomUp applyLink
+  where
+    applyLink :: Pandoc.Inline -> Pandoc.Inline
+    applyLink l@(Pandoc.Link attr inl (url, title)) = case parseShortcut $ T.pack url of
+        Right (shortcut, opt, txt) ->
+            case useShortcutFrom shortcuts shortcut opt $ fromMaybe (checkTitle inl) txt of
+                Success link   -> Pandoc.Link attr inl (T.unpack link, title)
+                Warning _ link -> Pandoc.Link attr inl (T.unpack link, title)
+                Failure _      -> l
```

Comment:
I wonder how can we report those errors in nice way :thinking: 
Maybe there exist some way in the `pandoc` or `hakyll` library to report errors. But this requires more research. Probably deserves separate issue :slightly_smiling_face:

Reply:
I've created an issue to track this: https://github.com/kowainik/hakyll-shortcut-links/issues/6